pyscf/neo/ase.py

```diff
@@ -86,52 +101,85 @@ def calculate(self, atoms=None, properties=['energy', 'forces', 'dipole'],
+            mf.mf_elec.xc = self.parameters.xc
+            if self.parameters.atom_grid is not None:
+                mf.mf_elec.grids.atom_grid = self.parameters.atom_grid
+            if self.parameters.add_vv10:
+                mf.mf_elec.nlc = 'VV10'
+                mf.mf_elec.grids.prune = None
+                mf.mf_elec.nlcgrids.atom_grid = (50,194)
+                mf.mf_elec.nlcgrids.prune = dft.gen_grid.sg1_prune
+            if isinstance(self.parameters.init_guess, str):
+                if self.parameters.init_guess != 'last_step' and self.parameters.init_guess != 'elec_only':
+                    mf.init_guess = self.parameters.init_guess
+            if self.parameters.conv_tol is not None:
+                mf.conv_tol = self.parameters.conv_tol
+            if self.parameters.conv_tol_grad is not None:
+                mf.conv_tol_grad = self.parameters.conv_tol_grad
+            if self.parameters.add_d3:
+                if not DFTD3_AVAILABLE:
+                    raise RuntimeError('DFTD3 PySCF extension not available')
+                mf.mf_elec = dftd3.dftd3(mf.mf_elec)
+            if self.parameters.add_solvent:
+                mf.scf(cycle=0) # TODO: remove this
+                mf = mf.ddCOSMO()
+
+            # always use dict as the initial guess
+            if isinstance(self.parameters.init_guess, dict):
+                dm0 = self.parameters.init_guess
+                mf.scf(dm0=[dm0['e']]+dm0['n'])
+            else:
+                mf.scf()
+            self.mf = mf
         else:
-            mf = neo.CDFT(mol, unrestricted=True, epc=self.parameters.epc)
-        mf.mf_elec.xc = self.parameters.xc
-        if self.parameters.atom_grid is not None:
-            mf.mf_elec.grids.atom_grid = self.parameters.atom_grid
-        if self.parameters.add_vv10:
-            mf.mf_elec.nlc = 'VV10'
-            mf.mf_elec.grids.prune = None
-            mf.mf_elec.nlcgrids.atom_grid = (50,194)
-            mf.mf_elec.nlcgrids.prune = dft.gen_grid.sg1_prune
-        if self.parameters.init_guess is not None:
-            mf.init_guess = self.parameters.init_guess
-        if self.parameters.conv_tol is not None:
-            mf.conv_tol = self.parameters.conv_tol
-        if self.parameters.conv_tol_grad is not None:
-            mf.conv_tol_grad = self.parameters.conv_tol_grad
-        if self.parameters.add_d3:
-            if not DFTD3_AVAILABLE:
-                raise RuntimeError('DFTD3 PySCF extension not available')
-            mf.mf_elec = dftd3.dftd3(mf.mf_elec)
-        if self.parameters.add_solvent:
-            mf.scf(cycle=0) # TODO: remove this
-            mf = mf.ddCOSMO()
+            # if mf is there, reset it with new mol
+            self.mf.reset(mol)
+            if self.parameters.init_guess == 'last_step':
+                self.mf.scf(dm0=[self.rdm1] + self.dm_nuc)
+            elif self.parameters.init_guess == 'elec_only':
+                self.mf.init_guess = self.rdm1
```

Comment:
This is wrong. You should pass `rdm1` to `scf`. `mf.init_guess` only take string arguments.

Reply:
No, this is correct. See line 49
https://github.com/pyscf/pyscf/blob/master/examples/scf/15-initial_guess.py